eth-custodian/test/EthCustodianProxy.js

```diff
@@ -0,0 +1,266 @@
+    kind: 'struct', fields: [
+      ['amount', 'u128'],
+      ['recipient', [20]],
+      ['ethCustodian', [20]],
+    ]
+  }
+};
+
+describe('EthCustodianProxy contract', () => {
+    let ethCustodian;
+    let ethCustodianProxy;
+    let nearProver;
+
+    let adminAccount;
+    let ethRecipientOnNear;
+    let user1;
+    let user2;
+
+    const nearEvmAccount = Buffer.from('v1.eth-connector.testnet');
+    const newProofProducerData = Buffer.from('new-producer.testnet');
+    const migrationBlock = 19672697;
+
+    beforeEach(async () => {
+        [adminAccount, ethRecipientOnNear, user1, user2] = await ethers.getSigners();
+
+        const nearProverMockContractFactory = await ethers.getContractFactory('NearProverMock')
+        nearProver = await nearProverMockContractFactory
+            .connect(adminAccount)
+            .deploy();
+
+        const ethCustodianContractFactory = await ethers.getContractFactory('EthCustodian');
+        ethCustodian = await ethCustodianContractFactory
+            .connect(adminAccount)
+            .deploy(
+                nearEvmAccount,
+                nearProver.address,
+                0,
+                adminAccount.address,
+                UNPAUSED_ALL);
+
+        const ethCustodianProxyContractFactory = await ethers.getContractFactory('EthCustodianProxy');
+        ethCustodianProxy = await upgrades.deployProxy(
+            ethCustodianProxyContractFactory,
+            [ethCustodian.address]
+        );
+
+        const nominateTx = await ethCustodian.nominateAdmin(ethCustodianProxy.address);
+        await nominateTx.wait();
+    
+        const acceptTx = await ethCustodian.acceptAdmin(ethCustodianProxy.address);
+        await acceptTx.wait();
+
+        const pauseTx = await ethCustodianProxy.pauseImpl(PAUSED_ALL);
+        await pauseTx.wait();
+    });
+
+    describe('EthCustodian', () => {
+        it('Should be paused', async () => {
+            const paused = await ethCustodian.paused();
+
+            expect(paused & PAUSED_DEPOSIT_TO_EVM).to.not.equal(0);
+            expect(paused & PAUSED_DEPOSIT_TO_NEAR).to.not.equal(0);
+            expect(paused & PAUSED_WITHDRAW).to.not.equal(0);
+        });
+    });
+
+    describe('Deposit', () => {
+        it('to EVM Should change the balance and emit the event', async () => {
+            const amountToTransfer = 50000;
+            const balanceBefore = ethers.BigNumber.from(
+                await ethers.provider.getBalance(ethCustodian.address));
+
+            const protocolMessage = nearEvmAccount + ':' + String(ethRecipientOnNear.address);
+            const options = { value: amountToTransfer };
+            await expect(
+                ethCustodianProxy
+                    .connect(user1)
+                    .depositToEVM(ethRecipientOnNear.address, options)
+            )
+                .to.emit(ethCustodian, 'Deposited')
+                .withArgs(ethCustodianProxy.address, protocolMessage, amountToTransfer, 0);
```

Comment:
It will work. For the proof we are only interested in the receiver. And then indexer will need to be changed to find the actual initiator of the transfer